subsys/shell/modules/app_service.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/printk.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <kernel_internal.h>
+#if defined(CONFIG_LOG_RUNTIME_FILTERING)
+#include <zephyr/logging/log_ctrl.h>
+#endif
+
+#include <malloc.h>
+#include <stdlib.h>
+#include <string.h>
+
+static int cmd_app_heap(const struct shell *sh, size_t argc, char **argv)
+{
+	ARG_UNUSED(sh);
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	/* Note: This doesn't use shell_print, so, might not work with all
+	 *       backends.
+	 */
+	malloc_stats();
```

Comment:
Would be kind of nice to have `mallinfo()` support, even though IIRC, it's a gnu-ism.

Actually, the fact that `malloc_stats()` does not use the `shell_print()` is a good enough reason to add `mallinfo()` support.

Reply:
Yes, in fact my initial code was based on `mallinfo2` but latest realized it's not supported and had to resort to `malloc_stats`.

Reply:
You could add it if you like. I would approve, likely also Keith

Reply:
Will take a stab, thanks.

Reply:
Just had to include `REQUIRES_FULL_LIBC` and it worked.